remote/remote_state_s3.go

```diff
@@ -117,6 +117,17 @@ func (s3Initializer S3Initializer) NeedsInitialization(remoteState *RemoteState,
 		return false, nil
 	}
 
+	// Nowadays it only makes sense to set the "dynamodb_table" attribute as it has
+	// been supported in Terraform since the release of version 0.10. The deprecated
+	// "lock_table" attribute is either set to NULL in the state file or missing
+	// from it altogether. Display a deprecation warning when the "lock_table"
+	// attribute is being used.
+	if util.KindOf(remoteState.Config["lock_table"]) == reflect.String && remoteState.Config["lock_table"] != "" {
+		terragruntOptions.Logger.Printf("%s\n", lockTableDeprecationMessage)
+		remoteState.Config["dynamodb_table"] = remoteState.Config["lock_table"]
+		delete(remoteState.Config, "lock_table")
```

Comment:
just to understand better, this change means now we're going to look at the `remoteState.config`

Reply:
Actually, this is doing the exact same thing as before within `configValuesEqual`, because go maps pass by reference. I am just moving the routine outside of the function to:

- make the map manipulation on `remoteState.Config` more explicit.
- ensure that the `configValuesEqual` function is side effect free.